xml/src/main/java/org/ehcache/xml/ConfigurationParser.java

```diff
@@ -408,7 +408,7 @@ public ListenersConfig listenersConfig() {
 
           @Override
           public Iterable<ServiceConfiguration<?>> serviceConfigs() {
-            Map<Class<? extends ServiceConfiguration>, ServiceConfiguration<?>> configsMap =
```

Comment:
Why?  Although the code works with the looser type, the tighter type is more correct imho.

Reply:
Since it's a local variable and I can't type `Class<? extends ServiceConfiguration<?>>` since it causes other parts of the code to scream

Reply:
Given that it's getClass() returning the erased type here and thus prevents us from doing the 'right' thing... wouldn't it make more sense to suppress the warning and leave the code as is?

Reply:
Ok